core/src/main/java/de/arbeitsagentur/opdt/keycloak/cassandra/event/persistence/entities/AdminEventEntity.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2024 Phase Two, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package de.arbeitsagentur.opdt.keycloak.cassandra.event.persistence.entities;
+
+import com.datastax.oss.driver.api.mapper.annotations.CqlName;
+import com.datastax.oss.driver.api.mapper.annotations.Entity;
+import com.datastax.oss.driver.api.mapper.annotations.PartitionKey;
+import lombok.*;
+
+@EqualsAndHashCode(of = {"id"})
+@Builder(toBuilder = true)
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+@CqlName("admin_events")
+public class AdminEventEntity {
+  @PartitionKey private String id;
```

Comment:
I think it would be benefical to use (realm_id, id) as partition key and time as clustering column (since you have a lot of queries where you only want events from/up to a specific timestamp)